io/byteslice_external_test.go

```diff
@@ -21,23 +21,29 @@ func TestByteSlice(t *testing.T) {
 
 // TestStutter tests what happens if the network stutters (split one message into several network packages).
 func TestStutter(t *testing.T) {
-	var values = []byte{0, 1, 2, 3, 4, 5, 6, 255}
+	values := []byte{0, 1, 2, 3, 4, 5, 6, 255}
 	r, w := io.Pipe()
 
+	werr := make(chan error, 1)
 	go func() {
 		for _, v := range values {
-			w.Write([]byte{v})
+			if _, err := w.Write([]byte{v}); err != nil {
+				werr <- err
```

Comment:
I just realize that this will get stuck on a full channel if there's a second error (which is likely after a first)
```suggestion
				werr <- err
				return
```

But even if we fix it this way, we'd still have to deal with the blocking `decodedValue.Decode(r)` call... It would also just get stuck and time out. You can use `context/test.AssertTerminatesQuickly` to fix that one. You can call `assert.NoError` in the function that you pass. Only calling `require` is forbidden in goroutines afair.

Reply:
Ahh, for that reason, you can also call `assert` in the goroutine and don't need the `werr` channel after all.

Reply:
Ok yes i also forgot that `assert` is thread safe :see_no_evil:  
Updated it.

Reply:
`AssertTerminatesQuickly` is in the last commit.  
@sebastianst